### PR TITLE
New Resource: `azurerm_restore_point_collection`

### PR DESCRIPTION
**Pull request by mbfrahry**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR adds support for Restore Point Collections will allow us to support restore points in the future

```
--- PASS: TestAccRestorePointCollection_basic (253.58s)
--- PASS: TestAccRestorePointCollection_update (243.54s)
```

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* New Resource: `azurerm_restore_point_collection` [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/compute/restore_point_collection_resource.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := restorepointcollections.RestorePointCollection{
+				Location: config.Location,
```
Comment (catriona-m) [inline internal/services/compute/restore_point_collection_resource.go:94]:
```suggestion
				Location: location.Normalize(config.Location),
```
should we normalize the location here?

File: internal/services/compute/restore_point_collection_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestnw-%[1]d"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "internal-%[1]d"
```
Comment (catriona-m) [inline internal/services/compute/restore_point_collection_resource_test.go:123]: should we prefix this with `acctest`?

File: website/docs/r/restore_point_collection.html.markdown [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+* `id` - The ID of the Restore Point Collection.
+
+## Timeouts
+
+The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/language/resources/syntax#operation-timeouts) for certain actions:
+
+* `create` - (Defaults to 45 minutes) Used when creating the Restore Point Collection.
+* `update` - (Defaults to 45 minutes) Used when updating the Restore Point Collection.
+* `delete` - (Defaults to 45 minutes) Used when deleting the Restore Point Collection.
+* `read` - (Defaults to 5 minutes) Used when retrieving the Restore Point Collection.
```
Comment (catriona-m) [inline website/docs/r/restore_point_collection.html.markdown:114]:
```suggestion
* `create` - (Defaults to 30 minutes) Used when creating the Restore Point Collection.
* `update` - (Defaults to 30 minutes) Used when updating the Restore Point Collection.
* `delete` - (Defaults to 30 minutes) Used when deleting the Restore Point Collection.
* `read` - (Defaults to 5 minutes) Used when retrieving the Restore Point Collection.
```

File: internal/services/compute/restore_point_collection_resource.go [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+			client := metadata.Client.Compute.RestorePointCollectionsClient
+
+			id, err := restorepointcollections.ParseRestorePointCollectionID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id, restorepointcollections.DefaultGetOperationOptions())
+			if err != nil {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
```
Comment (catriona-m) [inline internal/services/compute/restore_point_collection_resource.go:166]:
```suggestion
				return fmt.Errorf("retrieving %s: %+v", id, err)
```

**Review by catriona-m — APPROVED** [review]
Thanks @mbfrahry - I spotted a couple of minor things that might need fixed up but otherwise this LGTM!

**Comment by mbfrahry** [comment]
Thanks @catriona-m, taken care of

**Pull request merged**